content-based-message-routing/main.bal

```diff
@@ -0,0 +1,88 @@
+import ballerina/http;
+import ballerina/log;
+
+type Patient record {|
+    string name;
+    string dob;
+    string ssn;
+    string address;
+    string phone;
+    string email;
+|};
+
+type ReservationRequest record {|
+    Patient patient;
+    string doctor;
+    string hospital_id?;
+    string hospital;
+    string appointment_date;
+|};
+
+type Doctor record {|
+    string name;
+    string hospital;
+    string category;
+    string availability;
+    float fee;
+|};
+
+type ReservationResponse record {|
+    int appointmentNumber;
+    Doctor doctor;
+    Patient patient;
+    float fee;
+    string hospital;
+    boolean confirmed;
+    string appointmentDate;
+|};
+
+configurable int port = 8290;
+
+final http:Client grandOaksEP = check initializeHttpClient("http://localhost:9090/grandoaks/categories");
+final http:Client clemencyEP = check initializeHttpClient("http://localhost:9090/clemency/categories");
+final http:Client pineValleyEP = check initializeHttpClient("http://localhost:9090/pinevalley/categories");
+
+function initializeHttpClient(string url) returns http:Client|error => new (url);
+
+service /healthcare on new http:Listener(port) {
+    resource function post categories/[string category]/reserve(
+            @http:Payload ReservationRequest reservationRequest
+        ) returns ReservationResponse|http:NotFound|http:InternalServerError? {
+
+        http:Client hospitalEP = grandOaksEP; // default
+
+        string hospital = reservationRequest.hospital;
+        match hospital {
+            "grand oak community hospital" => {
```

Comment:
Can't we match by `hospital_id`? Is it really optional?

Reply:
That's OK IMO, ID makes more sense, right?

Reply:
Fixed via https://github.com/ballerina-guides/integration-tutorials/pull/4/commits/763a650a3b68fabf8f4bcf9c46a41a7ad2bc6871